### ISSUE TITLE
TagManager missing when creating a Matomo instance via composer create-project matomo/matomo 

### ISSUE THREAD

**Issue by bluedogmilan** [piwik/piwik#20140]
## Expected Behavior
Installling Matomo via composer should ship the TagManager plugin or notify about next step to get the complete installation

## Current Behavior
When creating a new Matomo instance via `composer create-project matomo/matomo`
the directory ./plugins/TagManager is empty.

```
total 8.0K
drwxrwxr-x  2 ubuntu ubuntu 4.0K Dec 27 10:31 .
drwxrwxr-x 95 ubuntu ubuntu 4.0K Dec 27 10:31 ..
```

## Environment
Distro:

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
```
PHP:
```
PHP 8.1.11 (cli) (built: Sep 29 2022 22:29:14) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.11, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.11, Copyright (c), by Zend Technologies
```
Composer:
```
Composer version 2.4.2 2022-09-14 16:11:15
```

**Comment by mikkeschiren** [comment]
That would never work - composer doesn't handle git sub modules. TagManager needs to be published as package, which it's not, and added as a dependency in composer.json for matomo. 
I am no fan of git sub modules at all - and the recommended approach for matomo should be to remove all sub modules and add them as packages instead.

**Comment by bluedogmilan** [comment]
My 2 cents are on a post install script that clones all the sub modules:
[https://getcomposer.org/doc/articles/scripts.md](url)

**Comment by mikkeschiren** [comment]
True, you could use a "post-create-project-cmd" event in composer.json that checks out the git sub modules. Still, I do think they shouldn't be git sub modules, at least they should be on packagist, so you could use whatever approach to install them.

**Comment by sgiehl** [comment]
I don't think setting up Matomo using composer create-project ever worked as expected. It's not only the submodule plugins, but also any other plugin.
I would personally discontinue and remove the package from composer, as it would require some more effort to get that running correctly I guess.
